src/etc/inc/config.lib.inc

```diff
@@ -138,7 +138,7 @@ function parse_config($parse = false) {
 	if (!$parse) {
 		if (file_exists($g['tmp_path'] . '/config.cache')) {
 			$config = unserialize(file_get_contents($g['tmp_path'] . '/config.cache'));
-			if (is_null($config)) {
+			if (!$config) {
```

Comment:
It could even do
if (!is_array($config))
because if $config is not an array then there is trouble.